### PR TITLE
Implement a new Metric Type which will be used for TableMetrics

### PR DESCRIPTION
**Pull request by vmeka2020**
JIRA:https://issues.apache.org/jira/browse/PHOENIX-6379

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetricImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (yanxinyi) [inline phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetricImpl.java:1]: format is not right

Comment (vmeka2020) [reply phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetricImpl.java:1]: Modified

**Comment by dbwong** [comment]
@mihir6692 Might want to review this.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 23s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m  7s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 48s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m 18s |  phoenix-core in 4.x has 945 extant spotbugs warnings.  |
| -0 :warning: |  patch  |   3m 24s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  5s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 29s |  phoenix-core: The patch generated 27 new + 0 unchanged - 0 fixed = 27 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 29s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 183m  8s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   | 223m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1138 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux be7528d71a99 4.15.0-126-generic #129-Ubuntu SMP Mon Nov 23 18:53:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 883170e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/1/testReport/ |
| Max. process+thread count | 4798 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by yanxinyi** [comment]
@vmeka2020 can you squash your commits and raise a new PR with the format of PHOENIX-number Jira title? It will attract to the Jira and run the Hadoop QA, thanks

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 40s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m  8s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 49s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m 14s |  phoenix-core in 4.x has 945 extant spotbugs warnings.  |
| -0 :warning: |  patch  |   3m 21s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  3s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  3s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 26s |  phoenix-core: The patch generated 21 new + 0 unchanged - 0 fixed = 21 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 181m 22s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   | 217m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1138 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 4b6477c0f31c 4.15.0-126-generic #129-Ubuntu SMP Mon Nov 23 18:53:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 883170e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/2/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/2/testReport/ |
| Max. process+thread count | 5080 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1138/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**